### PR TITLE
feat(astro): introduce `@unocss/astro` integration 

### PR DESCRIPTION
**Pull request by Dunqing**
close: #1402

Just insert the following code into the *.astro page or layout component.

```html
<script>import 'uno.css'</script>.
```

### Note: 

The imported 'uno.css' must be included on all pages or layouts that use `unocss`, otherwise it will not take effect.

I'm not sure if this is the best solution, but it should be the simplest one that requires no changes at this time.

### CODE REVIEW

File: examples/astro/src/components/Card.astro [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	.link-card {
+		list-style: none;
+		display: flex;
+		padding: 0.15rem;
+		background-image: var(--link-gradient);
+		background-size: 400%;
+		border-radius: 0.5rem;
+		background-position: 100%;
+		transition: background-position 0.6s cubic-bezier(0.22, 1, 0.36, 1);
+	}
```
Comment (antfu) [inline examples/astro/src/components/Card.astro:36]: Could we remove all these inline styles to use UnoCSS exclusively?

File: examples/astro/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Welcome to [Astro](https://astro.build)
```
Comment (antfu) [inline examples/astro/README.md:1]: We might remove the default readme, but instead, explain about how to integrate UnoCSS intro Astro and with the caveats you mentioned.

File: examples/astro/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    // Enable top-level await, and other modern ESM features.
+    "target": "ESNext",
+    "module": "ESNext",
+    // Enable node-style module resolution, for things like npm package imports.
+    "moduleResolution": "node",
+    // Enable JSON imports.
+    "resolveJsonModule": true,
+    // Enable stricter transpilation for better output.
+    "isolatedModules": true,
+    // Astro will directly run your TypeScript code, no transpilation needed.
```
Comment (antfu) [inline examples/astro/tsconfig.json:12]: We could remove these comments

**Review by antfu — CHANGES_REQUESTED** [review]

**Comment by antfu** [comment]
Thanks for the initiative. I have extended this PR to also introduce a new Astro integration so the requirement of importing `uno.css` is no longer a case.

**Review by antfu — APPROVED** [review]

**Comment by Dunqing** [comment]
> Thanks for the initiative. I have extended this PR to also introduce a new Astro integration so the requirement of importing `uno.css` is no longer a case.

Thank you, this is the better solution!

**Comment by aspiiire** [comment]
With the integration it doesn't build for me, it gives:
```
[unocss:global:build:generate] [unocss] does not found CSS placeholder in the generated chunks
This is likely an internal bug of unocss vite plugin
```
If I remove it from the integration and import unocss/vite on astro.config.mjs under vite.plugins and add import 'uno.css' everything works fine why?

**Pull request merged**